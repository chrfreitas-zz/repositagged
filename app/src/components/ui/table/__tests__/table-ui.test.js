import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Table from '../index';

describe('Table view', () => {
  it('should match with snapshot', () => {
    const header = ['Repository', 'Description', 'Language', 'Tags', ''];

    const repositories = [
      {
        id: 1,
        name: 'kubernetes',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s',
        language: 'go',
        tag: '',
      },
      {
        id: 2,
        name: 'django',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s',
        language: 'python',
        tag: '',
      },
      {
        id: 3,
        name: 'jest',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s',
        language: 'javascript',
        tag: '',
      },
    ];

    const renderer = new ShallowRenderer();
    const result = renderer.render(
      <Table header={header} body={repositories} />
    );
    expect(result).toMatchSnapshot();
  });
});
