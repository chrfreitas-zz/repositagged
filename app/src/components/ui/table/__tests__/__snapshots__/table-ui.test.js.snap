// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table view should match with snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Repository
      </th>
      <th>
        Description
      </th>
      <th>
        Language
      </th>
      <th>
        Tags
      </th>
      <th>
        
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        kubernetes
      </td>
      <td>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s
      </td>
      <td>
        go
      </td>
      <td />
      <td>
        <Button
          onClick={[Function]}
        >
          edit
        </Button>
      </td>
    </tr>
    <tr>
      <td>
        django
      </td>
      <td>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s
      </td>
      <td>
        python
      </td>
      <td />
      <td>
        <Button
          onClick={[Function]}
        >
          edit
        </Button>
      </td>
    </tr>
    <tr>
      <td>
        jest
      </td>
      <td>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s
      </td>
      <td>
        javascript
      </td>
      <td />
      <td>
        <Button
          onClick={[Function]}
        >
          edit
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;
